### ISSUE TITLE
keadm init fail, Error: CustomResourceDefinition.apiextensions.k8s.io "devices.devices.kubeedge.io" is invalid: spec.versions: Invalid value:

### ISSUE THREAD

**Issue by yookacn** [kubeedge/kubeedge#3811]
**What happened**:
Please help, keadm init fail, error information:
 keadm init --kubeedge-version=1.7.1 --advertise-address=192.168.8.101
Kubernetes version verification passed, KubeEdge installation will start...
Error: CustomResourceDefinition.apiextensions.k8s.io "devices.devices.kubeedge.io" is invalid: spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion{apiextensions.CustomResourceDefinitionVersion{Name:"v1alpha2", Served:true, Storage:true, Schema:(*apiextensions.CustomResourceValidation)(0xc00cedc4f0), Subresources:(*apiextensions.CustomResourceSubresources)(nil), AdditionalPrinterColumns:[]apiextensions.CustomResourceColumnDefinition(nil)}}: per-version schemas may not all be set to identical values (top-level validation should be used instead)
Usage:
  keadm init [flags]

Examples:

keadm init

- This command will download and install the default version of KubeEdge cloud component

keadm init --kubeedge-version=1.5.0  --kube-config=/root/.kube/config

  - kube-config is the absolute path of kubeconfig which used to secure connectivity between cloudcore and kube-apiserver

Flags:
      --advertise-address string   Use this key to set IPs in cloudcore's certificate SubAltNames field. eg: 10.10.102.78,10.10.102.79
      --domainname string          Use this key to set domain names in cloudcore's certificate SubAltNames field. eg: www.cloudcore.cn,www.kubeedge.cn
  -h, --help                       help for init
      --kube-config string         Use this key to set kube-config path, eg: $HOME/.kube/config (default "/root/.kube/config")
      --kubeedge-version string    Use this key to download and use the required KubeEdge version
      --master string              Use this key to set K8s master address, eg: http://127.0.0.1:8080
      --tarballpath string         Use this key to set the temp directory path for KubeEdge tarball, if not exist, download it

F0425 16:36:15.054066   17451 keadm.go:27] CustomResourceDefinition.apiextensions.k8s.io "devices.devices.kubeedge.io" is invalid: spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion{apiextensions.CustomResourceDefinitionVersion{Name:"v1alpha2", Served:true, Storage:true, Schema:(*apiextensions.CustomResourceValidation)(0xc00cedc4f0), Subresources:(*apiextensions.CustomResourceSubresources)(nil), AdditionalPrinterColumns:[]apiextensions.CustomResourceColumnDefinition(nil)}}: per-version schemas may not all be set to identical values (top-level validation should be used instead)
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0x1)
        /home/data/k8s/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:996 +0x8a
k8s.io/klog/v2.(*loggingT).output(0x2a8cc20, 0x3, {0x0, 0x0}, 0xc0001d5960, {0x1f19ef5, 0xc0005b4140}, 0xc0007eff10, 0x0)
        /home/data/k8s/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:945 +0x51d
k8s.io/klog/v2.(*loggingT).printDepth(0x200000003, 0x1a0, {0x0, 0x0}, 0x4117d1, {0xc0007eff60, 0x1, 0x1})
        /home/data/k8s/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:718 +0x132
k8s.io/klog/v2.(*loggingT).print(...)
        /home/data/k8s/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:703
k8s.io/klog/v2.Fatal(...)
        /home/data/k8s/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:1436
main.main()
        /home/data/k8s/kubeedge/kubeedge/keadm/cmd/keadm/keadm.go:27 +0x65

goroutine 19 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x0)
        /home/data/k8s/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:1131 +0x6a
created by k8s.io/klog/v2.init.0
        /home/data/k8s/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:416 +0xf4

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
command step:
$ wget https://github.com/kubeedge/kubeedge/releases/download/v1.7.1/keadm-v1.7.1-linux-amd64.tar.gz
$ tar -xvf keadm-v1.7.1-linux-amd64.tar.gz
$ cd keadm-v1.7.1-linux-amd64/keadm
$ cp keadm /usr/local/bin/
$ keadm init --kubeedge-version=1.7.1 --advertise-address=192.168.8.101

v1.6.2、v1.7.2、v1.8.0 has tried both, reporting the same error.

**Environment**:
- Kubernetes version (use `kubectl version`):  Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-01-18T23:30:10Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}
  Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.17", GitCommit:"f3abc15296f3a3f54e4ee42e830c61047b13895f", GitTreeState:"clean", BuildDate:"2021-01-13T13:13:00Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):  NAME="CentOS Linux" VERSION="7 (Core)" ID="centos" ID_LIKE="rhel fedora" VERSION_ID="7" PRETTY_NAME="CentOS Linux 7 (Core)"
  - Kernel (e.g. `uname -a`):  Linux kubeedge-cloud 3.10.0-1160.59.1.el7.x86_64
  - Go version (e.g. `go version`):  go1.17.4
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`): 
  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>

**Comment by RyanZhaoXB** [comment]
hello.  based on your error logs, it can be determined that the problem is in the CRD version.  your problem is the same as this issue (https://github.com/kubernetes/kubernetes/issues/82443 ). it is a version mapping issue. kubernetes provides a new version, which is apiextensions.k8s.io/v1, since kubernetes 1.17 and the defination of CRD is changed (https://kubernetes.io/zh/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#specify-multiple-versions).  and kubeedge has published the new version of device CRD in 1.9.

the suggestions are as follows:
1. use the latest version of kubeedge and keadm, at least >= 1.9
2. if you insist on using kubeedge1.7, please check these items and provide the result. i will help you to find a way out :)

```
1. kubectl api-resources |grep Custom
2. kubectl api-versions |grep apiextensions
3. cat devices_v1alpha2_device.yaml
```

**Comment by yookacn** [comment]
@RyanZhaoXB， thanks，least >v1.9 install successfly!
